src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Dao;
+import mate.academy.lib.Inject;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Dao
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User foundedUser = userService.findByEmail(email)
+                .orElseThrow(() -> new AuthenticationException(String.format(
+                        "User with login: %s is not exist", email)));
+
+        String inputPassword = HashUtil.hashPassword(password, foundedUser.getSalt());
+
+        if (foundedUser.getPassword().equals(inputPassword)) {
+            return foundedUser;
+        }
+
+        throw new AuthenticationException(String.format("Incorrect password: %s", inputPassword));
```

Comment:
Create only one condition for throwing `AuthenticationException` in `login()` method. You may combine two checks: whether the user has been found by login and does passwords match.